packages/framework/src/Model/Customer/User.php

```diff
@@ -140,13 +140,18 @@ public function __construct(
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Customer\UserData $userData
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerPasswordService $customerPasswordService
      */
-    public function edit(UserData $userData)
+    public function edit(UserData $userData, CustomerPasswordService $customerPasswordService)
     {
         $this->firstName = $userData->firstName;
         $this->lastName = $userData->lastName;
         $this->pricingGroup = $userData->pricingGroup;
         $this->telephone = $userData->telephone;
+
+        if ($userData->password !== null) {
+            $customerPasswordService->changePassword($this, $userData->password);
```

Comment:
You are right that it will be changed to `PasswordEncoderInterface` during phase 5. And since we are willing to finish the phase 5, I don't see a value of effort given to moving logic to `Facade`.